app/src/main/java/com/github/se/studybuddies/ui/chat/ChatScreen.kt

```diff
@@ -504,3 +597,115 @@ fun SendPhotoMessage(messageViewModel: MessageViewModel, showAddImage: MutableSt
         })
   }
 }
+
+@Composable
+fun SendLinkMessage(messageViewModel: MessageViewModel, showAddLink: MutableState<Boolean>) {
+  val linkState = remember { mutableStateOf("") }
+  val linkName = remember { mutableStateOf("") }
+
+  if (showAddLink.value) {
+    AlertDialog(
+        onDismissRequest = { showAddLink.value = false },
+        text = {
+          Box(
+              contentAlignment = Alignment.Center,
+              modifier = Modifier.padding(8.dp).fillMaxWidth()) {
+                OutlinedTextField(
+                    value = linkState.value,
+                    onValueChange = { linkState.value = it },
+                    modifier = Modifier.fillMaxWidth(),
+                    textStyle = TextStyle(color = Black),
+                    singleLine = true,
+                    placeholder = { Text(stringResource(R.string.enter_link)) },
+                )
+              }
+        },
+        confirmButton = {
+          SaveButton(linkState.value.isNotBlank()) {
+            val uriString = linkState.value.trim()
+            val uri =
+                if (!isValidUrl(uriString)) {
+                  Uri.parse("https://$uriString")
+                } else {
+                  Uri.parse(uriString)
+                }
+            linkName.value = uriString.substringAfter("//")
+
+            messageViewModel.sendLinkMessage(linkName.value, uri)
+            showAddLink.value = false
+            linkState.value = ""
+            linkName.value = ""
+          }
+        })
+  }
+}
+
+fun isValidUrl(url: String): Boolean {
+  return try {
+    val uri = Uri.parse(url)
+    uri.scheme == "http" || uri.scheme == "https"
+  } catch (e: Exception) {
+    false
+  }
+}
+
+@Composable
+fun SendFileMessage(messageViewModel: MessageViewModel, showAddFile: MutableState<Boolean>) {
+  val fileState = remember { mutableStateOf(Uri.EMPTY) }
+  val fileName = remember { mutableStateOf("") }
+  val context = LocalContext.current
+
+  val getContent =
+      rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri: Uri? ->
+        uri?.let { fileUri ->
+          fileState.value = fileUri
+          val cursor = context.contentResolver.query(fileUri, null, null, null, null)
+          cursor?.use {
+            if (it.moveToFirst()) {
+              val nameIndex = it.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+              if (nameIndex != -1) {
```

Comment:
You could remove one of the nested loop here, by directly affecting nameIndex in the if statement and checking != -1